src/commons/navigationBar/subcomponents/SicpNavigationBar.tsx

```diff
@@ -63,15 +98,412 @@ const SicpNavigationBar: React.FC = () => {
     usePortal: false
   };
 
+  const searchDrawerProps = {
+    onClose: handleCloseSearch,
+    autoFocus: true,
+    canEscapeKeyClose: true,
+    canOutsideClickClose: true,
+    enforceFocus: true,
+    hasBackdrop: true,
+    isOpen: isSearchOpen,
+    position: Position.LEFT,
+    usePortal: false
+  };
+
+  function queryIndexTrie(query: string) {
+    let node = indexTrie;
+    for (let i = 0; i < query.length; i++) {
+      const char = query[i];
+      if (node[char]) {
+        node = node[char];
+      } else {
+        return [];
+      }
+    }
+    if (node['value']) {
+      return node['value'];
+    }
+    return [];
+  }
+
+  function queryTextTrie(query: string) {
+    let node = textTrie;
+    for (let i = 0; i < query.length; i++) {
+      const char = query[i];
+      if (node[char]) {
+        node = node[char];
+      } else {
+        return [];
+      }
+    }
+    if (node['value']) {
+      return node['value'];
+    }
+    return [];
+  }
+
+  function autoComplete(str: string, limit: number, jsonData: any) {
+    if (str.length === 0) {
+      return [];
+    }
+    str = str.toLowerCase();
+    const trie = jsonData;
+
+    function next(node: any, i: number) {
```

Comment:
If we are using `any` like this, it's ideally only temporary, but more importantly, as it presents a lot of potential type safety issues, it should be aliased.

```ts
type TrieNode = any;
// Then, wherever `any` refers to a node in the trie...
function next(node: TrieNode, i: number) {
  // ...
}
```

This will not only greatly add readability and prevent bugs, but also developer experience (from completions, etc).